OpenRA.Mods.Common/Activities/Air/FlyOffMap.cs

```diff
@@ -35,6 +35,14 @@ public override bool Tick(Actor self)
 			if (IsCanceling || !self.World.Map.Contains(self.Location))
 				return true;
 
+			// VTOLs must take off first if they're not at cruise altitude
+			if (aircraft.Info.VTOL && self.World.Map.DistanceAboveTerrain(aircraft.CenterPosition) != aircraft.Info.CruiseAltitude)
+			{
+				QueueChild(new TakeOff(self));
+				return false;
+			}
+
+			// Using FlyTick directly is necessary to be able to exit the activity as soon as we leave the map
```

Comment:
Actually not true (!)

Activities can set `ChildHasPriority = false` in the constructor to turn off the logic where their children tick before they do. We could queue the `TakeOff` and `Fly` from `OnFirstRun` and simplify `Tick` down to the outside map check followed by `TickChild(self);`